modules/core/02-client/types/msgs.go

```diff
@@ -265,6 +269,32 @@ func (msg *MsgRecoverClient) ValidateBasic() error {
 	return nil
 }
 
+// NewMsgProvideCounterparty creates a new MsgProvideCounterparty instance
+func NewMsgProvideCounterparty(signer, clientID, counterpartyID string) *MsgProvideCounterparty {
+	return &MsgProvideCounterparty{
+		Signer:         signer,
+		ClientId:       clientID,
+		CounterpartyId: counterpartyID,
+	}
+}
+
+// ValidateBasic performs basic checks on a MsgProvideCounterparty.
+func (msg *MsgProvideCounterparty) ValidateBasic() error {
```

Comment:
We should probably validate the merkle path prefix as well. Unclear what exactly we should check yet. Let's make an issue for it so we don't lose track of it.

Reply:
Also should be testing this function